cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMirrorMaker2AssemblyOperator.java

```diff
@@ -0,0 +1,265 @@
+        String host = getServiceName(mirrorMaker2Name);
+        KafkaConnectApi apiClient = getKafkaConnectApi();
+        return apiClient.list(host, KafkaConnectCluster.REST_API_PORT).compose(deleteMirrorMaker2ConnectorNames -> {
+
+            MIRRORMAKER2_CONNECTORS.entrySet().stream()
+                    .forEach(connectorEntry -> deleteMirrorMaker2ConnectorNames.removeAll(mirrors.stream()
+                            .filter(mirror -> connectorEntry.getValue().apply(mirror) != null) // filter out non-existent connectors
+                            .map(mirror -> mirror.getSourceCluster() + "->" + mirror.getTargetCluster() + connectorEntry.getKey())
+                            .collect(Collectors.toSet())));
+            log.debug("{}: {}} cluster: delete mirror maker 2 connectors: {}", reconciliation, kind(), deleteMirrorMaker2ConnectorNames);
+            Stream<Future<Void>> deletionFutures = deleteMirrorMaker2ConnectorNames.stream()
+                    .map(connectorName -> apiClient.delete(host, KafkaConnectCluster.REST_API_PORT, connectorName));
+            Stream<Future<Void>> createUpdateFutures = mirrors.stream()
+                    .map(mirror -> reconcileMirrorMaker2Connectors(reconciliation, host, apiClient, kafkaMirrorMaker2, mirror));
+            return CompositeFuture.join(Stream.concat(deletionFutures, createUpdateFutures).collect(Collectors.toList())).map((Void) null);
+        });
+    }
+
+    private Future<Void> reconcileMirrorMaker2Connectors(Reconciliation reconciliation, String host, KafkaConnectApi apiClient, KafkaMirrorMaker2 mirrorMaker2, KafkaMirrorMaker2MirrorSpec mirror) {
+        String targetClusterAlias = mirror.getTargetCluster();
+        String sourceClusterAlias = mirror.getSourceCluster();
+        if (targetClusterAlias == null) {
+            return maybeUpdateMirrorMaker2Status(reconciliation, mirrorMaker2,
+                    new InvalidResourceException("targetCluster property is required"));
+        } else if (sourceClusterAlias == null) {
+            return maybeUpdateMirrorMaker2Status(reconciliation, mirrorMaker2,
+                    new InvalidResourceException("sourceCluster property is required"));
+        }
+        List<KafkaMirrorMaker2ClusterSpec> clusters = Optional.ofNullable(mirrorMaker2.getSpec().getClusters())
+            .orElse(Collections.emptyList());
+        Map<String, KafkaMirrorMaker2ClusterSpec> clusterMap = clusters.stream()
+            .filter(cluster -> targetClusterAlias.equals(cluster.getAlias()) || sourceClusterAlias.equals(cluster.getAlias()))
+            .collect(Collectors.toMap(KafkaMirrorMaker2ClusterSpec::getAlias, Function.identity()));
+
+        if (!clusterMap.containsKey(targetClusterAlias)) {
+            return maybeUpdateMirrorMaker2Status(reconciliation, mirrorMaker2,
+                    new InvalidResourceException("targetCluster with alias " + mirror.getTargetCluster() + " cannot be found in the list of clusters"));
```

Comment:
It's clearer to use a dotted path to where in the resource you're expecting to find the alias, i.e. "targetCluster with alias " + mirror.getTargetCluster() + " cannot be found in the list of clusters at spec.clusters".

Reply:
Done